Author: Modestas Vainius <modax@debian.org>
Description: fixes for Debian GNU/kFreeBSD
 kFreeBSD has the same userland as Linux. This change is necessary to enable
 kwsysProcessKill() to kill child processes on kFreeBSD. The bug was detected
 by CTestTestTimeout test.
 .
 https://buildd.debian.org/fetch.cgi?pkg=cmake;ver=2.8.1-1;arch=kfreebsd-amd64;stamp=1268849159
Origin: vendor
Forwarded: yes

--- a/Source/kwsys/ProcessUNIX.c
+++ b/Source/kwsys/ProcessUNIX.c
@@ -2373,7 +2373,8 @@
    Here we define the command to call on each platform and the
    corresponding parsing format string.  The parsing format should
    have two integers to store: the pid and then the ppid.  */
-#if defined(__linux__) || defined(__APPLE__) || defined(__FreeBSD__)
+#if defined(__linux__) || defined(__APPLE__) || defined(__FreeBSD__) \
+   || defined(__FreeBSD_kernel__)
 # define KWSYSPE_PS_COMMAND "ps axo pid,ppid"
 # define KWSYSPE_PS_FORMAT  "%d %d\n"
 #elif defined(__hpux) || defined(__sun__) || defined(__sgi) || defined(_AIX) \
